src/Components/People/EditPerson.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  CardContent,
+  CardActions,
+  TextField,
+  Typography,
+  Fab,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  SelectChangeEvent,
+} from "@mui/material";
+import DoneIcon from "@mui/icons-material/Done";
+import ClearIcon from "@mui/icons-material/Clear";
+import SaveIcon from "@mui/icons-material/Save";
+import { useEffect, useState } from "react";
+import { personEnabledDetails, PersonWithId } from "../../Interfaces/Person";
+import GroupServices from "../../Services/Groups";
+import { toast } from "react-toastify";
+import { IGroup } from "../../Interfaces/Group";
+
+const PersonEnabledDetails = (props: personEnabledDetails) => {
+  const { setAreSlotsEnabled, updatePerson, createPerson, person } = props;
+
+  const { id } = person;
+  const [newPerson, setNewPerson] = useState(person);
+  const [allGroups, setAllGroups] = useState();
+  const [group, setGroup] = useState("");
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setGroup(event.target.value);
+  };
+
+  useEffect(() => {
+    GroupServices.getAllGroups()
+      .then((res) => {
+        setAllGroups(
+          res.map((group: IGroup) => {
+            return (
+              <MenuItem key={group.id} value={group.id}>
+                {group.name}
+              </MenuItem>
+            );
+          })
+        );
+      })
+      .catch((err) => {
+        return toast.error(err.response.data);
+      });
+  }, []);
+
+  const editDetailTextBox = (
+    fieldName: string,
+    field: string,
+    mb: number = -1,
+    mt: number = 1,
+    ml: number = 2,
+    fontSize: number = 14
+  ) => {
+    return (
+      <TextField
+        InputProps={{ disableUnderline: true, style: { fontSize } }}
+        sx={{ mb, ml, mt }}
+        placeholder={fieldName}
+        variant="standard"
+        value={newPerson[field]}
+        onChange={(event) =>
+          setNewPerson((person: PersonWithId) => ({
+            ...person,
+            [field]: event.target.value,
+          }))
+        }
+      />
+    );
+  };
+
+  return (
+    <CardContent>
+      {editDetailTextBox("Name:", "name", 0.4, -0.5, 0, 16)}
```

Comment:
This could be done more nicely with [React Children Props](https://blog.logrocket.com/using-react-children-prop-typescript) for something like 
```javascript
<EditableTextBox name="color" label="Favorite color" size="small"/>
```